runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/WindmillTimeUtils.java

```diff
@@ -44,6 +44,10 @@ public class WindmillTimeUtils {
   public static Instant windmillToHarnessTimestamp(long timestampUs) {
     // Windmill should never send us an unknown timestamp.
     Preconditions.checkArgument(timestampUs != Long.MIN_VALUE);
+    // Set windmill min timestamp to beam min timestamp
+    if (timestampUs == Long.MIN_VALUE + 1) {
+      return BoundedWindow.TIMESTAMP_MIN_VALUE;
+    }
```

Comment:
actually would other values (ie Long.MIN_VALUE + 2) also be problematic (though that timestamp would be less likely to be observed)

maybe this should calculate result as before but then bounds check it with TIMESTAMP.MIN_VALUE similar to how it bound checks TIMESTAMP_MAX_VALUE